src/deno_dir.rs

```diff
@@ -313,31 +319,41 @@ impl DenoDir {
 
     let (output_code_filename, output_source_map_filename) =
       self.cache_path(&out.filename, &out.source_code);
-    let result =
-      self.load_cache(out.filename.as_str(), out.source_code.as_str());
-    match result {
-      Err(err) => {
-        if err.kind() == std::io::ErrorKind::NotFound {
-          Ok(out)
-        } else {
-          Err(err.into())
+    let mut maybe_output_code = None;
+    let mut maybe_source_map = None;
+
+    if !self.recompile {
+      let result =
+        self.load_cache(out.filename.as_str(), out.source_code.as_str());
+      match result {
+        Err(err) => {
+          if err.kind() == std::io::ErrorKind::NotFound {
+            return Ok(out);
+          } else {
+            return Err(err.into());
+          }
+        }
+        Ok((output_code, source_map)) => {
+          maybe_output_code = Some(output_code);
+          maybe_source_map = Some(source_map);
         }
       }
-      Ok((output_code, source_map)) => Ok(CodeFetchOutput {
-        module_name: out.module_name,
-        filename: out.filename,
-        media_type: out.media_type,
-        source_code: out.source_code,
-        maybe_output_code_filename: output_code_filename
-          .to_str()
-          .map(|s| s.to_string()),
-        maybe_output_code: Some(output_code),
-        maybe_source_map_filename: output_source_map_filename
-          .to_str()
-          .map(|s| s.to_string()),
-        maybe_source_map: Some(source_map),
-      }),
     }
+
+    Ok(CodeFetchOutput {
+      module_name: out.module_name,
+      filename: out.filename,
+      media_type: out.media_type,
+      source_code: out.source_code,
+      maybe_output_code_filename: output_code_filename
+        .to_str()
+        .map(|s| s.to_string()),
+      maybe_output_code: maybe_output_code,
+      maybe_source_map_filename: output_source_map_filename
+        .to_str()
+        .map(|s| s.to_string()),
```

Comment:
nit: could be simplified as `map(String::from)` (here and elsewhere)

Reply:
done ~